src/FrameDecorator/Page.php

```diff
@@ -546,6 +558,12 @@ function check_page_break(Frame $frame)
             }
 
             if ($next = $iter->get_prev_sibling()) {
+                while ($next->is_text_node() && $next->get_node()->nodeValue == "") {
+                    $next = $next->get_prev_sibling();
+                    if (!$next)
+                        break 2;
+                }
+
```

Comment:
This breaks with a whitespace node as first node, in which case it does not check for a parent in the following `if` block (because of the `break 2`).

Proposal:
```php
            $next = $iter->get_prev_sibling();
            // Skip empty text nodes
            while ($next && $next->is_text_node() && $next->get_node()->nodeValue === "") {
                $next = $next->get_prev_sibling();
            }

            if ($next) {
```